airflow-core/src/airflow/models/serialized_dag.py

```diff
@@ -207,26 +207,30 @@ def resolve_asset_uri_ref_dag_dep(self, dep_data: dict) -> Iterator[DagDependenc
 
     def resolve_asset_alias_dag_dep(self, dep_data: dict) -> Iterator[DagDependency]:
         dep_id = dep_data["dependency_id"]
-        for asset_id, asset_name in self.alias_names_to_asset_ids_names[dep_id]:
-            is_source_alias = dep_data["source"] == "asset-alias"
-            yield from [
-                # asset
-                DagDependency(
-                    source="asset" if is_source_alias else f"asset-alias:{dep_id}",
-                    target=f"asset-alias:{dep_id}" if is_source_alias else "asset",
-                    label=asset_name,
-                    dependency_type="asset",
-                    dependency_id=str(asset_id),
-                ),
-                # asset alias
-                DagDependency(
-                    source=f"asset:{asset_id}" if is_source_alias else dep_data["source"],
-                    target=dep_data["target"] if is_source_alias else f"asset:{asset_id}",
-                    label=dep_id,
-                    dependency_type="asset-alias",
-                    dependency_id=dep_id,
-                ),
-            ]
+        assets = self.alias_names_to_asset_ids_names[dep_id]
+        if assets:
+            for asset_id, asset_name in assets:
+                is_source_alias = dep_data["source"] == "asset-alias"
+                yield from [
+                    # asset
+                    DagDependency(
+                        source="asset" if is_source_alias else f"asset-alias:{dep_id}",
+                        target=f"asset-alias:{dep_id}" if is_source_alias else "asset",
+                        label=asset_name,
+                        dependency_type="asset",
+                        dependency_id=str(asset_id),
+                    ),
+                    # asset alias
+                    DagDependency(
+                        source=f"asset:{asset_id}" if is_source_alias else dep_data["source"],
+                        target=dep_data["target"] if is_source_alias else f"asset:{asset_id}",
+                        label=dep_id,
+                        dependency_type="asset-alias",
+                        dependency_id=dep_id,
+                    ),
+                ]
+        else:
+            yield DagDependency(**dep_data)
```

Comment:
Are we sure this is always going to work? I don’t see we blindly extracting `dep_data` elsewhere in the code base.

Reply:
> I thought we were going to reprase the dag?

Reserialization does not fix old dag versions (impossible).

Reply:
Got it. The latest version now should handle the dag_dep from 2.x now

Reply:
after a second thought, we probably will need to map `dataset` to `asset` for 2.x?

Reply:
This kind of things should probably be handled in a similar way to #49020.

Reply:
Ok, I'll take a look and create a separate PR for that

Reply:
Just notice the `dep_data` part.

Yep, what I meant is before it is written into databases. We store the data as DagDependency and then serialize it as dict before saving to db

https://github.com/apache/airflow/blob/88f9218eaf1f726d0f164fa5e46188dc0530d4ee/airflow-core/src/airflow/serialization/serialized_objects.py#L1674-L1679